ui/src/app/groupings/[groupingPath]/_components/export-dropdown.tsx

```diff
@@ -1,16 +1,129 @@
 'use client';
 
+import { useState, useRef } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+    faFileExport,
+    faCaretDown,
+    faUsers,
+    faUserPlus,
+    faUserMinus,
+    faIdCard
+} from '@fortawesome/free-solid-svg-icons';
+import { Tooltip, TooltipProvider, TooltipTrigger } from '@/components/ui/tooltip';
+import { getGroupingMembers } from '@/lib/actions';
 import { Button } from '@/components/ui/button';
-import { FileInput, ChevronDown } from 'lucide-react';
 
-const ExportDropdown = () => {
+interface Member {
+    lastName: string;
+    firstName: string;
+    uid: string;
+    uhUuid: string;
+}
+
+interface GroupData {
+    members: Member[];
+}
+
+const ExportDropdown = ({ groupPath }: { groupPath: string }) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const dropdownRef = useRef<HTMLDivElement>(null);
+
+    const memberTypes = [
+        { type: '', label: 'All Members', icon: faUsers },
+        { type: 'basis', label: 'Basis', icon: faIdCard },
+        { type: 'include', label: 'Include', icon: faUserPlus },
+        { type: 'exclude', label: 'Exclude', icon: faUserMinus }
+    ];
+
+    const fetchGroupData = async (memberType: string) => {
+        const fullPath = memberType === '' ? groupPath : `${groupPath}:${memberType}`;
+        const params = { page: undefined, size: undefined, sortString: 'name', isAscending: true };
+        const data: GroupData = await getGroupingMembers(fullPath, params);
+        exportFile(data, memberType);
+    };
+
+    const exportFile = (data: GroupData, memberType: string) => {
+        const members = data?.members;
+        const groupName = groupPath.split(':').pop();
+        const headers = ['Last', 'First', 'Username', 'UH Number', 'Email'];
+
+        const rows = members.map((member: Member) => {
+            let email = '';
+            if (member.uid && /^[a-zA-Z0-9._%+-]+$/.test(member.uid)) {
+                email = `${member.uid}@hawaii.edu`;
+            }
+
+            return [member.lastName, member.firstName, member.uid, member.uhUuid, email];
+        });
+
+        const csvContent = [headers, ...rows].map((row) => row.join(',')).join('\n');
+        const blob = new Blob([csvContent], { type: 'text/csv' });
+        const url = URL.createObjectURL(blob);
+        const link = document.createElement('a');
+        link.href = url;
+        link.download = memberType === '' ? `${groupName}_members_list.csv` : `${groupName}_${memberType}_list.csv`;
+        link.click();
+        URL.revokeObjectURL(url);
```

Comment:
Based on the coding best practice called single-use responsibility, this should be split into its own download() function.